src/main/scala/nl/vroste/zio/amqp/model/ExchangeType.scala

```diff
@@ -0,0 +1,22 @@
+package nl.vroste.zio.amqp.model
+
+sealed trait ExchangeType extends Product with Serializable {
```

Comment:
Is the `extends Product with Serializable` necessary?

Reply:
I believe it is used to have proper type inference in situations like [this](https://scastie.scala-lang.org/6Ejz7E8EQn61DVzRgZaqdg). Here is [question on so](https://stackoverflow.com/questions/36525804/scala-case-class-extending-product-with-serializable).